src/main/java/org/killbill/billing/catalog/api/VersionedCatalog.java

```diff
@@ -17,10 +17,37 @@
 
 package org.killbill.billing.catalog.api;
 
+import java.util.Date;
 import java.util.List;
 
-// Static catalog API methods must operate on the latest ("current") version
-public interface VersionedCatalog<CatalogType> extends StaticCatalog, Catalog {
+public interface VersionedCatalog {
+
+    /**
+     *
+     * @return All the (ordered) catalog versions
+     */
+    public List<StaticCatalog> getVersions();
+
+    /**
+     * The latest catalog version is the one used by the system when creating
+     *  or change the {@code Plan} associated with a subscription.
+     *
+     * @return the current (latest) catalog version.
+     */
+    public StaticCatalog getCurrentVersion();
+
+    /**
+     *
+     * @param targetDate the date used to find the catalog version
+     * @return the catalog version matching this date
+     *
+     *
+     * Assuming two StaticCatalog, S1 with a version date of D1, and S2 with a version date of D2:
```

Comment:
Nit: not 100% sure (build warnings would probably tell you), but I think the Javadocs description should be above the `@param` list.